thinc/util.py

```diff
@@ -583,6 +582,17 @@ def use_nvtx_range(message: str, id_color: int = -1):
         yield
 
 
+class time_context:
+    """Register the running time of a context."""
+
+    def __enter__(self):
+        self.start = time.perf_counter()
+        return self
+
+    def __exit__(self, type, value, traceback):
+        self.elapsed = time.perf_counter() - self.start
+
+
```

Comment:
Is this something that we need within `thinc` itself?

Reply:
couldn't it be useful sometimes? Would you prefer it in the speed-test file instead?

Reply:
Putting it in `thinc.util` means we're kind of committing to maintain this in the future as part of `thinc`, so I think it would be better to just have it in the script for now. Possibly in `thinc.tests.util` in the future if it's actually used in the test suite, but for now I think it's better in the script.